File: src/test/java/no/nav/fo/veilarbregistrering/registrering/bruker/SykmeldtRegistreringTestdataBuilder.java
package no.nav.fo.veilarbregistrering.registrering.bruker;

import no.nav.fo.veilarbregistrering.registrering.bruker.besvarelse.BesvarelseTestdataBuilder;

import java.time.LocalDateTime;

public class SykmeldtRegistreringTestdataBuilder {
    public static SykmeldtRegistrering gyldigSykmeldtRegistrering() {
        return new SykmeldtRegistrering()
                .setOpprettetDato(LocalDateTime.now())
                .setBesvarelse(BesvarelseTestdataBuilder.gyldigSykmeldtSkalTilbakeSammeJobbBesvarelse())
                .setTeksterForBesvarelse(TekstForSporsmalTestdataBuilder.gyldigeTeksterForSykmeldtBesvarelse());
    }
}

File: src/test/java/no/nav/fo/veilarbregistrering/registrering/bruker/SykmeldtRegistreringTestdataBuilder.kt
package no.nav.fo.veilarbregistrering.registrering.bruker

import no.nav.fo.veilarbregistrering.besvarelse.Besvarelse
import no.nav.fo.veilarbregistrering.besvarelse.BesvarelseTestdataBuilder
import java.time.LocalDateTime

object SykmeldtRegistreringTestdataBuilder {
    @JvmOverloads
    fun gyldigSykmeldtRegistrering(
        opprettetDato: LocalDateTime = LocalDateTime.now(),
        besvarelse: Besvarelse = BesvarelseTestdataBuilder.gyldigSykmeldtSkalTilbakeSammeJobbBesvarelse(),
    ): SykmeldtRegistrering {
        return SykmeldtRegistrering(
                opprettetDato = opprettetDato,
                besvarelse = besvarelse,
                teksterForBesvarelse = TekstForSporsmalTestdataBuilder.gyldigeTeksterForSykmeldtBesvarelse(),
        )
    }
}